gio/src/settings.rs

```diff
@@ -23,6 +35,68 @@ impl<O: IsA<Settings>> SettingsExtManual for O {
     fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError> {
         self.set_value(key, &value.to_variant())
     }
+
+    fn bind_with_mapping<P: IsA<glib::Object>>(
+        &self,
+        key: &str,
+        object: &P,
+        property: &str,
+        flags: SettingsBindFlags,
+        get_mapping: Option<Box<dyn Fn(&mut glib::Value, &glib::Variant) -> bool>>,
+        set_mapping: Option<Box<dyn Fn(&glib::Value, &glib::VariantType) -> glib::Variant>>,
+    ) {
```

Comment:
Maybe a builder here would be nicer (see `glib::Object::bind_property()`) so those optional functions only have to be provided if necessary.

Reply:
Another possible option is to split this into 3 functions, `bind_with_get_mapping()`, `bind_with_set_mapping()`, and `bind_with_get_set_mapping()`. (Both `None` should already be handled by the existing `bind()` function).

Reply:
I would remove the existing bind and add a builder (that also makes the flags optional)

Reply:
Looks good to me apart from this